### PR TITLE
Add version note regarding breaking changes in Azure Databricks 11.0

### PR DESCRIPTION
**Pull request by hallihan**
Add version note to mirror the note added to https://learn.microsoft.com/azure/architecture/databricks-monitoring/application-logs

Acknowledges: #187

### CODE REVIEW

**Review by skabou — APPROVED** [review]
🚀

File: README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Monitoring Azure Databricks in an Azure Log Analytics Workspace
 
+> Note:
+> This library supports Azure Databricks 10.x (Spark 3.2.x) and earlier (see [Supported configurations](#supported-configurations)). Azure Databricks 11.0 includes [breaking changes](https://learn.microsoft.com/en-us/azure/databricks/release-notes/runtime/11.0#log4j-is-upgraded-from-log4j-1-to-log4j-2) to the logging systems that the spark-monitoring library integrates with. The work required to update the spark-monitoring library to support Azure Databricks 11.0 (Spark 3.3.0) and newer is not currently planned.
```
Comment (ckittel) [inline README.md:4]: Formatting suggestion.  Take, leave, tweak as you see fit.

```suggestion
| :warning: | This library supports Azure Databricks 10.x (Spark 3.2.x) and earlier (see [Supported configurations](#supported-configurations)). Azure Databricks 11.0 includes [breaking changes](https://learn.microsoft.com/azure/databricks/release-notes/runtime/11.0#log4j-is-upgraded-from-log4j-1-to-log4j-2) to the logging systems that the spark-monitoring library integrates with. The work required to update the spark-monitoring library to support Azure Databricks 11.0 (Spark 3.3.0) and newer is not currently planned. |
|-----------|:--------------------------|
```

Comment (ckittel) [inline README.md:4]: If not going down the table route above...

```suggestion
> Note:
> This library supports Azure Databricks 10.x (Spark 3.2.x) and earlier (see [Supported configurations](#supported-configurations)). Azure Databricks 11.0 includes [breaking changes](https://learn.microsoft.com/azure/databricks/release-notes/runtime/11.0#log4j-is-upgraded-from-log4j-1-to-log4j-2) to the logging systems that the spark-monitoring library integrates with. The work required to update the spark-monitoring library to support Azure Databricks 11.0 (Spark 3.3.0) and newer is not currently planned.
```

**Review by ckittel — APPROVED** [review]
Feel free to apply either of these, or none of them :)

**Pull request merged**